client/src/components/data/SignUpData.ts

```diff
@@ -0,0 +1,5 @@
+export interface SignUpData {
+    name: string
+    email: string
+    password: string
+}
```

Comment:
Is there a need to separate this data into a different interface than the props? Personally I think merging them is fine, but it's up to you

Reply:
I was thinking we might need this when making requests to the server later on, ideally the UI state is separate from the model state